src/main/java/com/epam/esm/dao/impl/GiftCertificateDaoImpl.java

```diff
@@ -0,0 +1,154 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.GiftCertificateDAO;
+import com.epam.esm.dao.request.CertificateSearchCriteria;
+import com.epam.esm.dao.service.PersistenceService;
+import com.epam.esm.dao.sort.SortType;
+import com.epam.esm.model.Tag;
+import com.google.common.base.CaseFormat;
+import com.epam.esm.model.GiftCertificate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import javax.persistence.*;
+import javax.persistence.criteria.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+@EntityScan(basePackages = "com.epam.esm.model")
+@ComponentScan(basePackages = "com.epam.esm.dao.service.impl")
+public class GiftCertificateDaoImpl implements GiftCertificateDAO {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    private static final boolean ACTIVE_CERTIFICATE = true;
+    private static final boolean DELETED_CERTIFICATE = false;
+
+    private final PersistenceService<GiftCertificate> persistenceService;
```

Comment:
What is persistenceService? repository should not call service layer

Reply:
It's not service layer, this class contain methods that works with DB. Renamed this class to *PersistenceManager*